### DISCUSSION TITLE
prCreation non-pending makes no sense for repositories with at least one status triggered on pr event

### DISCUSSION THREAD

**Discussion by ns-ggeorgiev** [renovatebot/renovate#17453] [Suggest an Idea]
When I was reading the configuration documentation I was convinced that non-pending is a safe way to go, and I set the prCreation to non-pending.

It took me some time to finally realize that non-pending for github repositories with pr checks makes no sense.

Github statuses and checks - that run only after pr is created, and when prCreation is set to non-pending - prs are never created and status is always pending. it becomes chicken and the egg problem. 

These took me some debugging into logs and the code to realize. It will be helpful if the documentation is not so aggressive to advertise non-pending, or alerts in some other way for this issue with github pr development repos.

...maybe the property should automatically flip to "immediate" from "non-pending" when there are required pull-request checks.

**Comment by rarkins** [comment]
That's not true. GitHub Actions can be configured to run on branch pushes.

Example: https://github.com/containerbase/buildpack/blob/main/.github/workflows/build-push.yaml

Here is an example from the same repo of the PR event, which is different: https://github.com/containerbase/buildpack/blob/main/.github/workflows/build-pr.yaml

Any CI can choose to run on `push` or `pull_request` events. Results are recorded against the git commit SHA and not the branch or PR directly.

**Reply by ns-ggeorgiev** [reply]
Yes, you are right. Statuses are per sha and they can run on any event to create a record.

The concern is still valid though if you have even a single required status that is triggered only on a pull_request, which nowadays seems to be the standard and true for probably 95% of the cases. In my company it is clean slate 100%

**Reply by viceice** [reply]
nope, i know many projects which use branch only or combined variants

**Reply by rarkins** [reply]
I think you mean pull_request, not push_request?

**Reply by ns-ggeorgiev** [reply]
Anyways, I think one sentence to warn about this would not hurt, obviously not something that will benefit me - I already payed the price and I am aware of the issue.

**Reply by viceice** [reply]
also this isn't only happen on github. same will happen on all other renovate supported platforms if CI only runs on PR with not-pending setting.

**Reply by ns-ggeorgiev** [reply]
I am not familiar, it is not for me to say, but yes, it make sense to have the same issue.

That seems to be making point for a comment. As of this comment:

> nope, i know many projects which use branch only or combined variants

I would think it is not a requirement everybody to be at risk for something to be addressed in the documentation.

**Reply by rarkins** [reply]
You are welcome to improve the documentation via PR in https://github.com/renovatebot/renovate/blob/main/docs/usage/configuration-options.md#prcreation and https://github.com/renovatebot/renovate/blob/main/docs/usage/configuration-options.md#automergetype (the same limitation applies to branch automerge)

**Reply by ns-ggeorgiev** [reply]
By the way I just noticed that the documentation does not explain `approval` option at all. What will be a good description for it, when I find time to work on it, I can add this too.
